bin/cheritest/cheritest.h

```diff
@@ -151,6 +153,41 @@ void	signal_handler_clear(int sig);
 #define CHERITEST_VERIFY(cond) \
 	CHERITEST_VERIFY2(cond, "%s", "\'" #cond "\' is FALSE!")
 
+#define CHERITEST_CHECK_EQ(type, fmt, a, b, a_str, b_str)	do {	\
+		type __a = (a);						\
+		type __b = (b);						\
+		CHERITEST_VERIFY2(__a == __b, "%s (" fmt ") == %s ("	\
+		    fmt ") failed!", a_str, __a, b_str, __b);		\
+	} while (0)
+
+#define CHERITEST_CHECK_EQ_BOOL(a, b)	\
+	CHERITEST_CHECK_EQ(_Bool, "%d", a, b, __STRING(a), __STRING(b))
+#define CHERITEST_CHECK_EQ_INT(a, b)	\
+	CHERITEST_CHECK_EQ(int, "0x%lx", a, b, __STRING(a), __STRING(b))
+#define CHERITEST_CHECK_EQ_LONG(a, b)	\
+	CHERITEST_CHECK_EQ(long, "0x%lx", a, b, __STRING(a), __STRING(b))
+#define CHERITEST_CHECK_EQ_SIZE(a, b)	\
+	CHERITEST_CHECK_EQ(size_t, "0x%zx", a, b, __STRING(a), __STRING(b))
+
+static inline void
+_cheritest_check_cap_eq(void *__capability a, void *__capability b,
+    const char *a_str, const char *b_str)
+{
+#define CHECK_CAP_ATTR(accessor, fmt)						\
+	CHERITEST_VERIFY2(accessor(a) == accessor(b),				\
+	    __STRING(accessor) "(%s) (" fmt ") == " __STRING(accessor)		\
+	    "(%s) (" fmt ") failed!", a_str, accessor(a), b_str, accessor(b))
+	CHECK_CAP_ATTR(cheri_getaddress, "0x%ld");
```

Comment:
```suggestion
	CHECK_CAP_ATTR(cheri_getaddress, "0x%lx");
```
and repeated several times below.